content/kubeone/master/guides/operating-system-manager/usage/_index.en.md

```diff
@@ -0,0 +1,145 @@
+    - name: "aws"
+  bootstrapConfig:
+    files:
+      - path: /opt/bin/bootstrap
+        permissions: 755
+        content:
+          inline:
+            encoding: b64
+            data: |
+              #!/bin/bash
+
+              apt update && apt install -y curl jq
+
+      - path: /etc/systemd/system/bootstrap.service
+        permissions: 644
+        content:
+          inline:
+            encoding: b64
+            data: |
+              [Install]
+              WantedBy=multi-user.target
+
+              [Unit]
+              Requires=network-online.target
+              After=network-online.target
+              [Service]
+              Type=oneshot
+              RemainAfterExit=true
+              ExecStart=/opt/bin/bootstrap
+
+    modules:
+      runcmd:
+        - systemctl restart bootstrap.service
+
+  provisioningConfig:
+    files:
+      - path: /opt/hello-world
+        permissions: 644
+        content:
+          inline:
+            encoding: b64
+            data: echo "hello world"
+```
+
+**NOTE:** This is just for demonstration purposes and the custom OSPs would most likely be more complicated than this.
+
+2. Create your infrastructure
+
+```bash
+terraform apply -var=initial_machinedeployment_operating_system_profile=osp-install-curl
+```
+
+3. Now create your cluster and make sure that you are using the custom addons folder.
+
+```yaml
+apiVersion: kubeone.k8c.io/v1beta2
+kind: KubeOneCluster
+versions:
+  kubernetes: "1.22.5"
+cloudProvider:
+  aws: {}
+addons:
+  enable: true
+  path: "./custom-addons"
+```
+
+**NOTE:** The path for addons can be anything. But make sure that the custom OSP exists in that directory.
+
+### For an existing machine deployments
+
+For existing Machine Deployments; edit the machine deployment and update `k8c.io/operating-system-profile` annotation with the required OSP name.
+
+{{% notice warning %}}
+Secrets and OSC against the Machine Deployment will be rotated right away when we update the annotation. But the existing Machine Deployments will not be rotated automatically.
+Existing machine deployment should be rotated manually, for them to consume configurations generated by OSM.
+{{% /notice %}}
+
+To perform this rotation for existing Machine Deployments please follow the guide at [Rolling Restart MachineDeploments][rolling-restart].
+
+[rolling-restart]: {{< ref "../../../cheat_sheets/rollout_machinedeployment" >}}
+
+### For new machine deployments
+
+Create a new Machine Deployment and make sure that you add the `k8c.io/operating-system-profile` annotation with the required OSP name.
+
+## Additional Information
+
+- If an `OperatingSystemProfile` is updated, that wouldn't result in an automatic rotation of the machines. This is an intentional design decision since an `OperatingSystemProfile` can be associated with multiple Machine Deployments and it is the user's responsibility to rotate the machines when the `OperatingSystemProfile` is updated.
```

Comment:
```suggestion
- An update to `OperatingSystemProfile` will not result in an automatic rotation of the machines. This is an intentional design decision since an `OperatingSystemProfile` can be associated with multiple Machine Deployments. It is the user's responsibility to rotate the machines when the `OperatingSystemProfile` is updated.
```